
/EkfCalNode:
    ros__parameters:
        debug_log_level: 2
        data_logging_on: true
        body_data_rate: 100.0
        sim_params:
            seed: 0.0
            use_seed: true
            max_time: 10.0
            no_errors: false
            number_of_runs: 2
            run_number: 0
            truth_type: "cyclic"
            stationary_time: 0.0
            # pos_frequency: [0.0, 0.0, 0.0]
            pos_frequency: [0.3, 0.5, 0.7]
            # ang_frequency: [0.3, 0.4, 0.5]
            ang_frequency: [0.0, 0.0, 0.0]
            pos_offset: [0.0, 0.0, 0.0]
            ang_offset: [0.0, 0.0, 0.0]
            ang_amplitude: 0.1
            pos_amplitude: 1.0

        filter_params:
            process_noise: [
                0.0, 0.0, 0.0, # Position
                0.0, 0.0, 0.0, # Velocity
                1.0e-3, 1.0e-3, 1.0e-3, # Acceleration
                0.0, 0.0, 0.0, # Angular Position
                1.0e-2, 1.0e-2, 1.0e-2, # Angular Velocity
                0.0, 0.0, 0.0, # Angular Acceleration
            ]

        imu_list:
            - "vn300"

        camera_list:
            - "ace"

        fiducial_list:
            - "charuco"

        imu:
            vn300:
                use_for_prediction: true
                is_extrinsic: false
                is_intrinsic: false
                rate: 400.0
                topic: "/vn300/imu_uncompensated"
                variance: [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
                pos_i_in_b: [0.0, 0.0, 0.0]
                ang_i_to_b: [1.0, 0.0, 0.0, 0.0]
                acc_bias: [0.0, 0.0, 0.0]
                omg_bias: [0.0, 0.0, 0.0]
                pos_stability: 1.0e-2
                ang_stability: 1.0e-2
                acc_bias_stability: 1.0e-3
                omg_bias_stability: 1.0e-3
                sim_params:
                    time_bias_error: 1.0e-3
                    time_skew_error: 1.0e-9
                    time_error: 1.0e-6
                    pos_error: [0.0, 0.0, 0.0]
                    ang_error: [0.0, 0.0, 0.0]
                    acc_error: [1.0e-3, 1.0e-3, 1.0e-3]
                    omg_error: [1.0e-2, 1.0e-2, 1.0e-2]
                    acc_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    omg_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]

        camera:
            ace:
                rate: 20.0
                topic:  "/my_camera/pylon_ros2_camera_node/image_raw"
                pos_c_in_b: [0.0, 0.0, 0.0]
                ang_c_to_b: [0.5, -0.5, 0.5, -0.5]
                variance: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
                fiducial: "charuco"
                pos_stability: 1.0e-9
                ang_stability: 1.0e-9
                intrinsics:
                    F: 1.0
                    c_x: 320.0
                    c_y: 240.0
                    k_1: 0.0
                    k_2: 0.0
                    p_1: 0.0
                    p_2: 0.0
                    pixel_size: 1.0e-2
                sim_params:
                    time_bias_error: 1.0e-3
                    time_skew_error: 1.0e-9
                    time_error: 1.0e-6
                    pos_error: [1.0e-2, 1.0e-2, 1.0e-2]
                    ang_error: [1.0e-2, 1.0e-2, 1.0e-2]

        fiducial:
            charuco:
                fiducial_type: 1
                squares_x: 5
                squares_y: 7
                square_length: 0.04
                marker_length: 0.02
                pos_f_in_g: [5.0, 0.0, 0.0]
                ang_f_to_g: [1.0, 0.0, 0.0, 0.0]
                variance: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
                sim_params:
                    pos_error: [1.0e-3, 1.0e-3, 1.0e-3]
                    ang_error: [1.0e-2, 1.0e-2, 1.0e-2]
                    t_vec_error: [1.0e-3, 1.0e-3, 1.0e-3]
                    r_vec_error: [1.0e-2, 1.0e-2, 1.0e-2]